app/DynamicsAdapter/Fams3Adapter.Dynamics/Types/InformationSourceType.cs

```diff
@@ -16,4 +18,18 @@ protected InformationSourceType(int value, string name) : base(value, name)
         {
         }
     }
+    public static class StringExtension
+    {
+        public static int? SuppliedByID (this string source )
```

Comment:
I really like the pattern but I would do it on ProviderProfile like 
```c#
public static int? SuppliedById(this ProviderProfile profile)
```
this would limit the usage of it, otherwise you can use it on any string

Reply:
makes sense.